File: Febuary/twoSum.py
'''
The Challenge: Given a list nums and a target, find two numbers that add up to the target and return their indices.

nums = [2, 7, 11], target = 9
Result: [0, 1] (because nums[0] + nums[1] = 9)

The Tool: Instead of seen = set(), we use prev_map = {}.

Key: The number.

Value: The index where we found it.
'''

def twoSums(nums, target):
    index_map = {}

    for i in range(len(nums)):
        current_num = nums[i]
        complement = target-current_num
        print(complement)

        if complement in index_map:
            return [index_map.values(complement), current_num]
        else:
            index_map[i] = nums[i]


nums = [2, 7, 11, 15]
target = 9

print(twoSums(nums=nums, target=target))

File: Febuary/twoSum.cpp
/*
Given an array of integers nums and an integer target, 
return indices of the two numbers such that they add up to target

Example 1:
Input: nums = [2,7,11,15], target = 9
Output: [0,1]
Explanation: Because nums[0] + nums[1] == 9, we return [0, 1].
*/

#include <iostream>
#include <vector>
#include <string>
using namespace std;


vector<int> twoSum(vector<int>& nums, int target) {
    vector<int> indices = {};

    

    return indices;
}

int main(){
    vector<int> nums = {2,7,11,15}; 
    int target = 9;

    vector<int> result = twoSum(nums, target);

    for (int num : result){
        cout << num << " ";
    }

    cout << endl;


    return 0;
}